arches/app/views/resource.py

```diff
@@ -651,17 +651,24 @@ def get(
             resource_instance = models.ResourceInstance.objects.get(pk=resourceid)
             edits = models.EditLog.objects.filter(resourceinstanceid=resourceid)
             permitted_edits = []
+            nodegroup_ids_from_edits = [
+                uuid.UUID(nodegroupid) if nodegroupid else nodegroupid
+                for nodegroupid in edits.values_list("nodegroupid", flat=True)
+            ]
+            nodegroups_for_edits = models.NodeGroup.objects.filter(
+                pk__in=nodegroup_ids_from_edits
+            ).in_bulk()
             for edit in edits:
                 if edit.nodegroupid is not None:
-                    if request.user.has_perm("read_nodegroup", edit.nodegroupid):
+                    edit_nodegroup = nodegroups_for_edits[uuid.UUID(edit.nodegroupid)]
```

Comment:
This could fail if the nodegroup was deleted. Let's use .get() and deal with the missing nodegroup case.